library/unwind/src/libunwind.rs

```diff
@@ -87,7 +87,7 @@ pub type _Unwind_Exception_Cleanup_Fn =
 // rustc_codegen_ssa::src::back::symbol_export, rustc_middle::middle::exported_symbols
 // and RFC 2841
 #[cfg_attr(
-    all(feature = "llvm-libunwind", any(target_os = "fuchsia", target_os = "linux")),
+    any(target_os = "fuchsia", all(feature = "llvm-libunwind", target_os = "linux")),
```

Comment:
Hm, this seems suspicious -- I would expect us to change bootstrap to enable llvm-libunwind by default for Fuchsia in config.rs (roughly), is there a reason that's not the approach used here?

Reply:
That's what I wanted to do originally, but it's not the way things were structured for other linux to begin with. Instead the meaning of "no" changes depending on the platform configuration being built.

```
# Use LLVM libunwind as the implementation for Rust's unwinder.
# Accepted values are 'in-tree' (formerly true), 'system' or 'no' (formerly false).
# This option only applies for Linux and Fuchsia targets.
# On Linux target, if crt-static is not enabled, 'no' means dynamic link to
# `libgcc_s.so`, 'in-tree' means static link to the in-tree build of llvm libunwind
# and 'system' means dynamic link to `libunwind.so`. If crt-static is enabled,
# the behavior is depend on the libc. On musl target, 'no' and 'in-tree' both
# means static link to the in-tree build of llvm libunwind, and 'system' means
# static link to `libunwind.a` provided by system. Due to the limitation of glibc,
# it must link to `libgcc_eh.a` to get a working output, and this option have no effect.
```

Untangling this doesn't sound like a particularly fun time tbh. I think I would have structured the config differently but that will probably require a breaking change and changelog entry.

That being said, I'm fine with doing it your preferred way for Fuchsia in this PR and letting a future PR bring Linux in line.

Reply:
Yeah, I agree it's a bit of a mess. I'd prefer we try to decrease it rather than increase it, and I think moving the platform-dependency into bootstrap is probably the way to go. Starting with Fuchsia and dealing with Linux (which as you say, is more complicated) later makes sense to me.

Reply:
Moved to this but had to migrate llvm-libunwind to be a per-target config (as it probably should be). I don't feel like breaking everyone in the process so kept a global option for now.